appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/Comparable.kt

```diff
@@ -0,0 +1,6 @@
+package com.bumble.appyx.interactions.core
+
+interface Comparable<in T> {
+    
+    fun isEqualTo(other: T): Boolean
+}
```

Comment:
Discussion memo:

1. Get rid of this interface on individual properties
2. On Props classes, rename this + the method such that it requires a `isRunning: StateFlow<Boolean>` which is a combined flow of all of the individual animatables `isRunning` flow, combining them with `or`
3. Distinct until changed on this flow: when it flips to `true` or `false`, we can trigger the animation started / finished callbacks automatically

Cleaner, works automatically, and adds no assumptions that can go wrong easily as with the equals check

Reply:
didn't work due to `isRunning` value not being backed by `mutableState` when animation start/finish 
https://android.googlesource.com/platform/frameworks/support/+/refs/heads/androidx-main/compose/animation/animation-core/src/commonMain/kotlin/androidx/compose/animation/core/Animatable.kt#299